Update to reflect FrozenTime

yosus opened the pull request:
Minor wordings update to reflect FrozenTime usage. Updated the examples with more outputs. 

`Day-month` using `31-01` to avoid format-confusion
`Hour:minute` using `22:11` to show 24 hour format

yosus committed:
Update to reflect FrozenTime

Minor wordings update to reflect FrozenTime usage. Updated the examples with more outputs. 

`Day-month` using `31-01` to avoid format-confusion
`Hour:minute` using `22:11` to show 24 hour format

garas commented on en/core-libraries/time.rst:
```diff
@@ -69,45 +71,55 @@ as a UNIX timestamp.
 ============
 
-Once created, you can manipulate ``FrozenTime`` instances using setter methods::
+Once created, you cannot manipulate ``FrozenTime`` instances as it is immutable. Using setter methods will not work::
```
Paragraph shouldn't start with _you cannot_.

Start with correct example which show you need to assign new instance to variable.

You can insert incorrect example later in paragraph.

othercorey replied:
This is misleading. It sounds like you can't call the methods.

othercorey commented on en/core-libraries/time.rst:
```diff
@@ -69,45 +71,55 @@ as a UNIX timestamp.
-    // Returns '2014-04-12 12:22:30'
-    $now = FrozenTime::parse('now');
+    // Outputs '1/31/21, 10:11 PM'
+    echo FrozenTime::parse('now');
```
These should output a specific format to match the one used in setTestNow.

yosus committed:
Updated Manipulation Part

Manipulation part updated based on conversations

yosus reviewed (commented):
Updated based on conversations

ndm2 commented on en/core-libraries/time.rst:
```diff
@@ -37,77 +37,92 @@ For more details on Chronos please see `the API documentation
-There are a few ways to create ``FrozenTime`` instances::
+``FrozenTime`` are immutable objects that useful when you want to prevent
+accidental changes to data, or when you want to avoid order based dependency
+issues. Refer to ``Time`` instances for mutable objects. There are a few ways to create ``FrozenTime`` instances::
```
```suggestion
``FrozenTime`` are immutable objects that are useful when you want to prevent
accidental changes to data, or when you want to avoid order based dependency
issues. Refer to ``Time`` instances for mutable objects.

There are a few ways to create ``FrozenTime`` instances::
```

ndm2 commented on en/core-libraries/time.rst:
```diff
@@ -37,77 +37,92 @@ For more details on Chronos please see `the API documentation
-    // Returns '2014-04-12 12:22:30'
+    // Outputs '2021-01-31 22:11:30'
     $now = FrozenTime::parse('now');
+	echo $now->i18nFormat('yyyy-MM-dd HH:mm:ss');
```
```suggestion
    echo $now->i18nFormat('yyyy-MM-dd HH:mm:ss');
```

ndm2 commented on en/core-libraries/time.rst:
```diff
@@ -37,77 +37,92 @@ For more details on Chronos please see `the API documentation
 ============
 
-Once created, you can manipulate ``FrozenTime`` instances using setter methods::
+Once created, ``FrozenTime`` instances cannot be manipulated as it is immutable. You will need to create another instance. You can also use the methods provided by PHP's built-in ``DateTime`` class::
```
```suggestion
Once created, ``FrozenTime`` instances will not change anymore, as they are immutable. Manipulating them will create new instances that need to be reassigned::
```

ndm2 commented on en/core-libraries/time.rst:
```diff
@@ -37,77 +37,92 @@ For more details on Chronos please see `the API documentation
+    echo $newTime->i18nFormat('yyyy-MM-dd HH:mm:ss');
 
-You can also use the methods provided by PHP's built-in ``DateTime`` class::
+Using setter methods to manipulate ``FrozenTime`` instances will not work::
```
```suggestion
You can also use the methods provided by PHP's built-in ``DateTime`` class::

    $time = $time->setDate(2013, 10, 31);

Failing to reassign the new ``FrozenTime`` instances will result in the original, unmodified instance being used::
```

ndm2 commented on en/core-libraries/time.rst:
```diff
@@ -37,77 +37,92 @@ For more details on Chronos please see `the API documentation
-    $now = $now->subDays(5)
+    $time = FrozenTime::create(2021, 1, 31, 22, 11, 30);
+    $newTime = $time->subDays(5)
+		->addHours(-2)
```
```suggestion
        ->addHours(-2)
```

ndm2 commented on en/core-libraries/time.rst:
```diff
@@ -37,77 +37,92 @@ For more details on Chronos please see `the API documentation
     // Using strtotime strings.
-    $now = $now->modify('+5 days');
+	$newTime = $time->modify('+1 month -5 days -2 hours');
+	// Outputs '2/26/21, 8:11 PM'
```
```suggestion
    $newTime = $time->modify('+1 month -5 days -2 hours');
    // Outputs '2/26/21, 8:11 PM'
```

ndm2 commented on en/core-libraries/time.rst:
```diff
@@ -116,15 +131,20 @@ Formatting
 
 This method sets the default format used when converting an object to json::
 
+    
```
```suggestion
```

ndm2 commented on en/core-libraries/time.rst:
```diff
@@ -116,15 +131,20 @@ Formatting
     FrozenDate::setJsonEncodeFormat(static function($time) {
         return $time->format(DATE_ATOM);
     });
+    
```
```suggestion
```

ndm2 commented on en/core-libraries/time.rst:
```diff
@@ -138,22 +158,22 @@ This method sets the default format used when converting an object to json::
 dates. CakePHP makes this a snap::
 
-    $now = Time::parse('2014-10-31');
+	$time = FrozenTime::parse('2021-01-31 22:11:30');
```
```suggestion
    $time = FrozenTime::parse('2021-01-31 22:11:30');
```

ndm2 commented on en/core-libraries/time.rst:
```diff
@@ -138,22 +158,22 @@ This method sets the default format used when converting an object to json::
-    // Prints a localized datetime stamp.
-    echo $now;
+	// Prints a localized datetime stamp. Outputs '1/31/21, 10:11 PM'
+	echo $time;
```
```suggestion
    // Prints a localized datetime stamp. Outputs '1/31/21, 10:11 PM'
    echo $time;
```

ndm2 commented on en/core-libraries/time.rst:
```diff
@@ -138,22 +158,22 @@ This method sets the default format used when converting an object to json::
-    // Outputs '10/31/14, 12:00 AM' for the en-US locale
-    $now->i18nFormat();
+    // Outputs '1/31/21, 10:11 PM' for the en-US locale
+	echo $time->i18nFormat();
```
```suggestion
    echo $time->i18nFormat();
```

ndm2 commented on en/core-libraries/time.rst:
```diff
@@ -138,22 +158,22 @@ This method sets the default format used when converting an object to json::
-    // Use the full date and time format
-    $now->i18nFormat(\IntlDateFormatter::FULL);
+    // Use the full date and time format. Outputs 'Sunday, January 31, 2021 at 10:11:30 PM Eastern Standard Time'
+	echo $time->i18nFormat(\IntlDateFormatter::FULL);
```
```suggestion
    echo $time->i18nFormat(\IntlDateFormatter::FULL);
```

ndm2 commented on en/core-libraries/time.rst:
```diff
@@ -164,8 +184,22 @@ https://unicode-org.github.io/icu/userguide/format_parse/datetime/#datetime-form
+    // Outputs 'Sunday, Jumada II 18, 1442 AH at 10:11:30 PM Eastern Standard Time'
+    echo $time->i18nFormat(\IntlDateFormatter::FULL, null, 'en-SA@calendar=islamic');
+    
+    
```
```suggestion
```

ndm2 commented on en/core-libraries/time.rst:
```diff
@@ -221,6 +265,10 @@ You can, however, modify this default at runtime::
+    
+    // Outputs '31 ene. 2021 22:11'
+    echo $time->nice(); 
+    
```
```suggestion
```

ndm2 commented on en/core-libraries/time.rst:
```diff
@@ -298,86 +349,97 @@ Comparing With the Present
-    // Within 2 next weeks.
-    echo $time->isWithinNext('2 weeks');
+    // Within 2 next weeks. Outputs 'true'
+     var_export($time->isWithinNext('2 weeks'));
```
```suggestion
    var_export($time->isWithinNext('2 weeks'));
```

ndm2 commented on en/core-libraries/time.rst:
```diff
@@ -298,86 +349,97 @@ Comparing With the Present
+    echo $newDate->format('Y-m-d H:i:s');
 
-Attempts to modify the timezone on a ``Date`` instance are also ignored::
+    
```
```suggestion
```

ndm2 commented on en/core-libraries/time.rst:
```diff
@@ -233,14 +281,18 @@ Likewise, it is possible to alter the default formatting string to be used for
 
-    // The same method exists on Date, FrozenDate and FrozenTime
-    Time::setToStringFormat([
+    // The same method exists on Date, FrozenDate, Time and FrozenTime
```
```suggestion
    // The same method exists on Date, FrozenDate, and Time
```

ndm2 commented on en/core-libraries/time.rst:
```diff
@@ -233,14 +281,18 @@ Likewise, it is possible to alter the default formatting string to be used for
 
-    // The same method exists on Date, FrozenDate and FrozenTime
-    Time::setToStringFormat('yyyy-MM-dd HH:mm:ss');
+    // The same method exists on Date, FrozenDate, Time and FrozenTime
```
```suggestion
    // The same method exists on Date, FrozenDate, and Time
```

ndm2 commented on en/core-libraries/time.rst:
```diff
@@ -252,29 +304,28 @@ Formatting Relative Times
-    $now = new Time('Aug 22, 2011');
-    echo $now->timeAgoInWords(
+    $time = new FrozenTime('Jan 31, 2021');
+    // On 12-Jun-21, outputs '4 months, 1 week, 6 days ago'
```
```suggestion
    // On June 12, 2021, this would output '4 months, 1 week, 6 days ago'
```

ndm2 commented on en/core-libraries/time.rst:
```diff
@@ -37,77 +37,92 @@ For more details on Chronos please see `the API documentation
-    $now->year(2013)
-        ->month(10)
-        ->day(31);
+    $time = FrozenTime::parse('2021-01-31 22:11:30');
+    
+    // Create another instance
+    $newTime = $time->setDate(2013, 10, 31);
```
```suggestion
    $time = FrozenTime::now();
    
    // Create and reassign a new instance
    $time = $now->year(2013)
        ->month(10)
        ->day(31);
```

yosus committed:
Update en/core-libraries/time.rst

Co-authored-by: Oliver Nowak <ndm2@users.noreply.github.com>

yosus committed:
Update en/core-libraries/time.rst

Co-authored-by: Oliver Nowak <ndm2@users.noreply.github.com>

yosus committed:
Update en/core-libraries/time.rst

Co-authored-by: Oliver Nowak <ndm2@users.noreply.github.com>

yosus committed:
Update en/core-libraries/time.rst

Co-authored-by: Oliver Nowak <ndm2@users.noreply.github.com>

yosus committed:
Update en/core-libraries/time.rst

Co-authored-by: Oliver Nowak <ndm2@users.noreply.github.com>

yosus committed:
Update en/core-libraries/time.rst

Co-authored-by: Oliver Nowak <ndm2@users.noreply.github.com>

yosus committed:
Update en/core-libraries/time.rst

Co-authored-by: Oliver Nowak <ndm2@users.noreply.github.com>

yosus committed:
Update en/core-libraries/time.rst

Co-authored-by: Oliver Nowak <ndm2@users.noreply.github.com>

yosus committed:
Update en/core-libraries/time.rst

Co-authored-by: Oliver Nowak <ndm2@users.noreply.github.com>

yosus committed:
Update en/core-libraries/time.rst

Co-authored-by: Oliver Nowak <ndm2@users.noreply.github.com>

yosus committed:
Update en/core-libraries/time.rst

Co-authored-by: Oliver Nowak <ndm2@users.noreply.github.com>

yosus committed:
Update en/core-libraries/time.rst

Co-authored-by: Oliver Nowak <ndm2@users.noreply.github.com>

yosus committed:
Update en/core-libraries/time.rst

Co-authored-by: Oliver Nowak <ndm2@users.noreply.github.com>

yosus committed:
Update en/core-libraries/time.rst

Co-authored-by: Oliver Nowak <ndm2@users.noreply.github.com>

yosus committed:
Update en/core-libraries/time.rst

Co-authored-by: Oliver Nowak <ndm2@users.noreply.github.com>

yosus committed:
Update en/core-libraries/time.rst

Co-authored-by: Oliver Nowak <ndm2@users.noreply.github.com>

yosus committed:
Update en/core-libraries/time.rst

Co-authored-by: Oliver Nowak <ndm2@users.noreply.github.com>

yosus committed:
Update en/core-libraries/time.rst

Co-authored-by: Oliver Nowak <ndm2@users.noreply.github.com>

yosus committed:
Update en/core-libraries/time.rst

Co-authored-by: Oliver Nowak <ndm2@users.noreply.github.com>

yosus committed:
Minor change to text

Modified based on suggestions @ https://github.com/cakephp/docs/pull/7038#discussion_r651701226

yosus reviewed (commented):
minor change to text based on [https://github.com/cakephp/docs/pull/7038#discussion_r651701226](https://github.com/cakephp/docs/pull/7038#discussion_r651701226)

othercorey commented:
@markstory

markstory commented on en/core-libraries/time.rst:
```diff
@@ -37,77 +37,100 @@ For more details on Chronos please see `the API documentation
 .. start-time
 
-Creating Time Instances
+Creating FrozenTime Instances
 =======================
```
Title underlines need to be the same length as the text above them.

markstory commented on en/core-libraries/time.rst:
```diff
@@ -298,86 +352,96 @@ Comparing With the Present
-    echo $time->isThisWeek();
-    echo $time->isThisMonth();
-    echo $time->isThisYear();
+    var_export($time->isYesterday());
+    var_export($time->isThisWeek());
+    var_export($time->isThisMonth());
+    var_export($time->isThisYear());
```
```suggestion
    debug($time->isYesterday());
    debug($time->isThisWeek());
    debug($time->isThisMonth());
    debug($time->isThisYear());
```

I don't think `var_export` is what we want here as we're not trying to create executable code. `debug()` will also print the true/false properly and is more inline with the rest of the cake docs.

markstory commented on en/core-libraries/time.rst:
```diff
@@ -298,86 +352,96 @@ Comparing With the Present
-    // Within 2 days.
-    echo $time->isWithinNext(2);
+    // Within 2 days. Outputs 'false'
+    var_export($time->isWithinNext('2 days'));
```
```suggestion
    debug($time->isWithinNext('2 days'));
```

markstory commented on en/core-libraries/time.rst:
```diff
@@ -298,86 +352,96 @@ Comparing With the Present
-    // Within 2 next weeks.
-    echo $time->isWithinNext('2 weeks');
+    // Within 2 next weeks. Outputs 'true'
+    var_export($time->isWithinNext('2 weeks'));
```
```suggestion
    debug($time->isWithinNext('2 weeks'));
```

markstory commented on en/core-libraries/time.rst:
```diff
@@ -298,86 +352,96 @@ Comparing With the Present
+    $time = new FrozenTime('-72 hours');
+    
+    // Within past 2 days. Outputs 'false'
+    var_export($time->wasWithinLast('2 days'));
```
```suggestion
    debug($time->wasWithinLast('2 days'));
```

markstory commented on en/core-libraries/time.rst:
```diff
@@ -298,86 +352,96 @@ Comparing With the Present
-    // Within past 2 weeks.
-    echo $time->wasWithinLast('2 weeks');
+    // Within past 3 days. Outputs 'true'
+    var_export($time->wasWithinLast('3 days'));
```
```suggestion
    debug($time->wasWithinLast('3 days'));
```

markstory commented on en/core-libraries/time.rst:
```diff
@@ -298,86 +352,96 @@ Comparing With the Present
+    var_export($time->wasWithinLast('3 days'));
+    
+    // Within past 2 weeks. Outputs 'true'
+    var_export($time->wasWithinLast('2 weeks'));
```
```suggestion
    debug($time->wasWithinLast('2 weeks'));
```

markstory commented on en/core-libraries/time.rst:
```diff
@@ -298,86 +352,96 @@ Comparing With the Present
 .. end-time
 
-Dates
+FrozenDate
 =====
```
Title underlines need to be the same length as the text above them.

othercorey commented on en/core-libraries/time.rst:
```diff
@@ -298,86 +352,96 @@ Comparing With the Present
-``Date`` is that ``Date`` does not track time components, and is always in UTC.
+The immutable ``FrozenDate`` class in CakePHP implements the same API and methods as
+:php:class:`Cake\\I18n\\FrozenTime` does. The main difference between ``FrozenTime`` and
+``FrozenDate`` is that ``FrozenDate`` does not track time components, and is always in UTC.
```
```suggestion
``FrozenDate`` is that ``FrozenDate`` does not track time components.
```
This is no longer true, but it's an implementation detail that shouldn't matter.

yosus replied:
ok, will remove it from next PR. Can't change it now cause already merged.

ndm2 commented on en/core-libraries/time.rst:
```diff
@@ -37,77 +37,100 @@ For more details on Chronos please see `the API documentation
 ============
 
-Once created, you can manipulate ``FrozenTime`` instances using setter methods::
+Once created, ``FrozenTime`` instances cannot be manipulated or changed anymore, as they are immutable. Manipulating them will create new instance::
```
This sounds like Schrödinger's time objects again, as if you can, and can not manipulate them.

yosus replied:
How about this  :

Once created, ``FrozenTime`` instances cannot be manipulated or changed, as it is immutable. A new instance can be created from a ``FrozenTime`` instance.

yosus committed:
Update en/core-libraries/time.rst

Co-authored-by: Mark Story <mark@mark-story.com>

yosus committed:
Update en/core-libraries/time.rst

Co-authored-by: Mark Story <mark@mark-story.com>

yosus committed:
Update en/core-libraries/time.rst

Co-authored-by: Mark Story <mark@mark-story.com>

yosus committed:
Update en/core-libraries/time.rst

Co-authored-by: Mark Story <mark@mark-story.com>

yosus committed:
Update en/core-libraries/time.rst

Co-authored-by: Mark Story <mark@mark-story.com>

yosus committed:
Update en/core-libraries/time.rst

Co-authored-by: Mark Story <mark@mark-story.com>
